src/utils/serializers.ts

```diff
@@ -9,18 +12,91 @@ interface BridgeId {
   depositCount: number;
 }
 
+interface CommonPendingTx {
+  networkId: number;
+  from: Chain;
+  to: Chain;
+  timestamp: number;
+  token: Token;
+  amount: BigNumber;
+}
+
+type PendingDepositTxData = {
+  status: "pending-deposit";
+  depositTxHash: string;
+};
+
+type PendingClaimTxData = {
+  status: "pending-claim";
+  depositTxHash: string;
+  claimTxHash: string;
+};
+
+type PendingDepositTx = CommonPendingTx & PendingDepositTxData;
+
+type PendingClaimTx = CommonPendingTx & PendingClaimTxData;
+
+export type PendingTx = PendingDepositTx | PendingClaimTx;
+
+interface CommonSerializedPendingTx {
+  networkId: number;
+  from: number;
+  to: number;
+  timestamp: number;
+  token: Token;
+  amount: string;
+}
+
+type SerializedPendingDepositTx = CommonSerializedPendingTx & PendingDepositTxData;
+
+type SerializedPendingClaimTx = CommonSerializedPendingTx & PendingClaimTxData;
+
+export type SerializedPendingTx = SerializedPendingDepositTx | SerializedPendingClaimTx;
+
 const bridgeIdParser = StrictSchema<SerializedBridgeId, BridgeId>()(
   z.string().transform((bridgeId): BridgeId => {
     const [networkIdString, depositCountString] = bridgeId.split("-");
     const networkId = z.number().parse(Number(networkIdString));
     const depositCount = z.number().parse(Number(depositCountString));
+
     return {
       networkId,
       depositCount,
     };
   })
 );
 
+const serializedPendingTxDepositParser = StrictSchema<SerializedPendingDepositTx>()(
+  z.object({
+    status: z.literal("pending-deposit"),
+    depositTxHash: z.string(),
+    networkId: z.number(),
+    from: z.number(),
+    to: z.number(),
+    timestamp: z.number(),
+    token: tokenParser,
+    amount: z.string(),
+  })
+);
+
+const serializedPendingTxClaimParser = StrictSchema<SerializedPendingClaimTx>()(
+  z.object({
+    status: z.literal("pending-claim"),
+    depositTxHash: z.string(),
+    claimTxHash: z.string(),
+    networkId: z.number(),
+    from: z.number(),
+    to: z.number(),
+    timestamp: z.number(),
+    token: tokenParser,
+    amount: z.string(),
+  })
+);
+
+const serializedPendingTxParser = StrictSchema<SerializedPendingTx>()(
+  z.union([serializedPendingTxDepositParser, serializedPendingTxClaimParser])
+);
+
 const deserializeBridgeId = (
```

Comment:
Which util?

Reply:
`deserializeBridgeId`

Reply:
I meant why not do this as we always do:

```diff
diff --git a/src/utils/serializers.ts b/src/utils/serializers.ts
index 7e7c0fd..f5f53db 100644
--- a/src/utils/serializers.ts
+++ b/src/utils/serializers.ts
@@ -97,12 +97,6 @@ const serializedPendingTxParser = StrictSchema<SerializedPendingTx>()(
   z.union([serializedPendingTxDepositParser, serializedPendingTxClaimParser])
 );

-const deserializeBridgeId = (
-  value: unknown
-): z.SafeParseReturnType<SerializedBridgeId, BridgeId> => {
-  return bridgeIdParser.safeParse(value);
-};
-
 const serializeBridgeId = (parsedBridgeId: BridgeId): SerializedBridgeId => {
   const { networkId, depositCount } = parsedBridgeId;
   return `${networkId}-${depositCount}`;
@@ -199,8 +193,8 @@ const deserializePendingTx = (
 };

 export {
+  bridgeIdParser,
   serializedPendingTxParser,
-  deserializeBridgeId,
   serializeBridgeId,
   serializePendingTx,
   deserializePendingTx,
diff --git a/src/views/bridge-details/bridge-details.view.tsx b/src/views/bridge-details/bridge-details.view.tsx
index 2bf209d..94561f6 100644
--- a/src/views/bridge-details/bridge-details.view.tsx
+++ b/src/views/bridge-details/bridge-details.view.tsx
@@ -23,7 +23,7 @@ import { AsyncTask, isMetamaskUserRejectedRequestError } from "src/utils/types";
 import { getBridgeStatus, getChainName, getCurrencySymbol } from "src/utils/labels";
 import { formatTokenAmount, formatFiatAmount, multiplyAmounts } from "src/utils/amounts";
 import { calculateTransactionResponseFee } from "src/utils/fees";
-import { deserializeBridgeId } from "src/utils/serializers";
+import { bridgeIdParser } from "src/utils/serializers";
 import { Bridge } from "src/domain";
 import routes from "src/routes";
 import { FIAT_DISPLAY_PRECISION, getEtherToken } from "src/constants";
@@ -123,7 +123,7 @@ const BridgeDetails: FC = () => {

   useEffect(() => {
     if (env && account.status === "successful") {
-      const parsedBridgeId = deserializeBridgeId(bridgeId);
+      const parsedBridgeId = bridgeIdParser.safeParse(bridgeId);
       if (parsedBridgeId.success) {
         const { depositCount, networkId } = parsedBridgeId.data;
         void getBridge({
```

We only create fractions that wrap parsers when we need some extra param in the parser. Otherwise, we just export and use the raw parser.

It's not very relevant. Do as you prefer.

Reply:
We can address this in the issue you opened about reviewing this serializers module as well